tests/test_error_tools.py

```diff
@@ -23,3 +24,13 @@ def test_convert_enametoolong(self):
                 raise exception
         self.assertEqual(ctx.exception.exc, exception)
         self.assertEqual(ctx.exception.path, "/tmp/test")
+
+    @unittest.skipIf(sys.platform != "win32", "requires Windows")
+    def test_convert_resourcelocked_windows(self):
+        exception = OSError(32, "resource locked")
```

Comment:
Ahh, digging into it a bit it looks like you might need to pass this `32` through as the `winerror` parameter to the `OSError` constructor? https://docs.python.org/3/library/exceptions.html#OSError

Reply:
Well that's... odd... Somehow that feels wrong to me.

Reply:
Probably because POSIX and Win32 use different error codes for different things? :shrug:

Reply:
No I meant the first argument should always be the error code, at least in theory. I don't know why if we pass the error code in as the first argument we don't get it back out as the first argument in `args`.

Reply:
Hmmm, that's a good point - I wonder if it suggests that the original code was wrong, and it should have actually been `if getattr(exc_value, "winerror", None) == 32:` ? :shrug:

Reply:
From the docs:

> Under Windows, this gives you the native Windows error code. The errno attribute is then an approximate translation, in POSIX terms, of that native error code.
>
> Under Windows, if the winerror constructor argument is an integer, the errno attribute is determined from the Windows error code, and the errno argument is ignored. On other platforms, the winerror argument is ignored, and the winerror attribute does not exist.

This sounds to me like we shouldn't be touching it, and if it does exist it won't be 32.

Reply:
Checking back through the history, this code was [originally added](https://github.com/PyFilesystem/pyfilesystem2/commit/b32e0ec1fbebe933af54ebe209d7fb67ef057b13) by @willmcgugan - I know it's unlikely after 5 years, but I wonder if he might remember what it was _supposed_ to be doing?

Although looking back even further it seems this code came from https://github.com/PyFilesystem/pyfilesystem/blob/master/fs/errors.py#L297 ? Which was [added in 2009](https://github.com/PyFilesystem/pyfilesystem/commit/de98e4d839784e12f5f7a015eac5cf8bdd3bd43a), oh my! :open_mouth: 
Probably needs someone to experiment with actually raising the exception on Windows, rather than trying to guess at "manually reproducing" the matching exception? I guess there's a possibility that this code may not even be needed now, given the changes to both Windows and Python over the past 12 years? :man_shrugging:

Reply:
Haven't a clue. I'm afraid I had to dump a lot of my long term memory to make space for new Javascript libraries.

Some of that code was deduced experimentally. I do remember seeing behaviour that did not match any docs I could find. It could very well be outdated now.

Reply:
Any other ideas?